src/Rules/Operators/InvalidComparisonOperationRule.php

```diff
@@ -53,6 +59,22 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
+		$result = $this->operatorTypeSpecifyingExtensionRegistryProvider->getRegistry()->callOperatorTypeSpecifyingExtensions(
+			$node,
+			$scope->getType($node->left),
+			$scope->getType($node->right),
```

Comment:
This might cause some issues below level 7, but I'm not sure how to improve it. The operator type specifying extensions don't support union well anyway.